sql/src/main/java/io/crate/planner/node/management/ExplainPlan.java

```diff
@@ -177,35 +176,48 @@ private Row buildResponse(Map<String, Long> apeTimings, Map<String, Map<String,
                                                                                    Collection<NodeOperation> nodeOperations) {
         Set<String> nodeIds = NodeOperationGrouper.groupByServer(nodeOperations).keySet();
 
-        if (nodeIds.size() > 0) {
-            CompletableFuture<Map<String, Map<String, Long>>> resultFuture = new CompletableFuture<>();
-            TransportCollectProfileOperation collectProfileOperation = getTransportCollectProfileOperation(executor, jobId);
-
-            ConcurrentHashMap<String, Map<String, Long>> mergedMap = new ConcurrentHashMap<>(nodeIds.size());
-            AtomicInteger counter = new AtomicInteger(nodeIds.size());
-
-            for (String nodeId : nodeIds) {
-                collectProfileOperation.collect(nodeId)
-                    .whenComplete((map, throwable) -> {
-                        if (throwable == null) {
-                            mergedMap.put(nodeId, map);
-                            if (counter.decrementAndGet() == 0) {
-                                resultFuture.complete(mergedMap);
-                            }
-                        } else {
-                            resultFuture.completeExceptionally(throwable);
-                        }
-                    });
-            }
-            return resultFuture;
-        } else {
-            // Collect from local JobExecutionContext
-            return executor
+        CompletableFuture<Map<String, Map<String, Long>>> resultFuture = new CompletableFuture<>();
+        TransportCollectProfileOperation remoteCollectOperation = getRemoteCollectOperation(executor, jobId);
+
+        ConcurrentHashMap<String, Map<String, Long>> timingsByNodeId = new ConcurrentHashMap<>(nodeIds.size());
+        boolean needsCollectLocal = !nodeIds.contains(executor.localNodeId());
+
+        AtomicInteger remainingCollectOps = new AtomicInteger(nodeIds.size());
+        if (needsCollectLocal) {
+            remainingCollectOps.incrementAndGet();
+        }
+
+        // Collect from remote JobExecutionContexts
+        for (String nodeId : nodeIds) {
+            remoteCollectOperation.collect(nodeId)
+                .whenComplete(mergeResultsAndCompleteFuture(resultFuture, timingsByNodeId, remainingCollectOps, nodeId));
+        }
+
+        // Also collect from local JobExecutionContext if needed
```

Comment:
Personally I'd remove both comments, it's a "what" comment that is just as apparent by reading the code.
If anything, a comment explaining why the local collect is handled special would make more sense. But imho not really necessary.

Reply:
You're totally right, the comment does not add any value. Will be removed.